sections/Chapter 3 Sprint 1/postgreSQL.tex

```diff
@@ -2,33 +2,92 @@ \subsection{Relational Databases}
+\cite[Chapter 1.3]{DBSBook}
 
-These transactions encapsulate several atomic operations and thus the transactions themselves are not inherently atomic which can cause data inconsistencies. 
-To prevent this from happening, a relational database must ensure four crucial properties:
-\begin{itemize}
+Compared to storing data on a regular file system, a database system provides many advantages including atomicity of operations, concurrent access to data, and lowered inconsistency and redundancy of data \cite[Chapter 1.2]{DBSBook}.
+These advantages can be directly seen in the ACID properties that databases adhere to when performing a transaction (see Section \ref{sec:SQL}) \cite[Chapter~17]{DBSBook}.
+\begin{itemize} \label{ACID}
     \item Atomicity: A transaction must either be fully completed or partial side-effects of a failed transaction must be undone.
     \item Consistency: A transaction in isolation must ensure values remain consistent after a transaction has been completed or terminated.
     \item Isolation: Transactions are unaware of other transactions being executed concurrently to avoid confusion.
     \item Durability: Changes caused by a committed transaction persist even in the event of system failures.
 \end{itemize}
 
-A relational database consists of several layers.
-The lowest level is the physical layer which describes how the data are stored physically.
-The purpose of the relational model is abstract over the physical layer of the database.
+In the coming sections, we will describe how one can model and describe a relational database design, both mathematically and with a more graphic design.
+After doing this, we will examine how operations on these models can be described using SQL (Section~\ref{sec:SQL}) and relational algebra (Section~\ref{sec:relationalAlgebra}).
 
-This abstraction is known as the logical layer and allows database administrators to manage the physical storage without directly manipulating the physical data representation.
-The logical layer describes what data are stored and the relationships between the data.
+\subsubsection*{Mathematical description}
+Relational database systems can be mathematically described using relations and sets, mapping a unique key to a tuple of information \cite[Chapter~2.3]{DBSBook}.
+The values of the tuples contained within the relation can be described by the attributes of the relation and their corresponding domains \cite{KatjaFirstPP}. 
+The relations are often described using a \textit{relational schema}, denoting the name and domains of the attributes.
 
-The highest of abstraction is the view layer which describes only part of the database. It exits to simplify the interaction with the system. Many views may exist for the same database.
+Equation \ref{eq:relational_schema} shows an example of a relation describing books as tuples containing three text fields and a positive integer.
```

Comment:
Maybe explain what the text fields and positive integer are supposed to model. Seems a little vague at the moment.

Reply:
The text fields and integers are arbitrarily chosen. One could just as well have used something else - I just chose text and numbers to show that we can model both.

Reply:
Then maybe add that they are arbitrarily chosen just to avoid confusion